src/main/java/net/ilexiconn/llibrary/client/model/tabula/animation/AnimationPlayerAnimator.java

```diff
@@ -0,0 +1,95 @@
+package net.ilexiconn.llibrary.client.model.tabula.animation;
+
+import net.ilexiconn.llibrary.client.model.tabula.ITabulaModelAnimator;
+import net.ilexiconn.llibrary.client.model.tabula.TabulaModel;
+import net.ilexiconn.llibrary.client.model.tabula.container.TabulaAnimationComponentContainer;
+import net.ilexiconn.llibrary.client.model.tabula.container.TabulaAnimationContainer;
+import net.ilexiconn.llibrary.client.model.tools.AdvancedModelRenderer;
+import net.ilexiconn.llibrary.server.animation.Animation;
+import net.ilexiconn.llibrary.server.animation.IAnimatedEntity;
+import net.ilexiconn.llibrary.server.animation.NamedAnimation;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.MathHelper;
+
+/**
+ * Tabula Model animator that can play Tabula animations. USE {@link NamedAnimation} !!
+ * @author jglrxavpok
+ * @param <T> the entity type, needs to be an {@link IAnimatedEntity}
+ */
+public class AnimationPlayerAnimator<T extends Entity&IAnimatedEntity> implements ITabulaModelAnimator<T> {
+    @Override
+    public void setRotationAngles(TabulaModel model, T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float rotationYaw, float rotationPitch, float scale) {
+        Animation anim = entity.getAnimation();
+        TabulaAnimationContainer animation;
+        if(anim instanceof NamedAnimation) {
+            animation = model.getAnimation(((NamedAnimation) anim).getName());
+        } else {
+            return;
+        }
+
+        if(animation == null)
+            return;
+
+        model.resetToDefaultPose();
+
+        preAnimationCallback(model, entity, limbSwing, limbSwingAmount, ageInTicks, rotationYaw, rotationPitch);
+
+        animation.getComponents().entrySet().forEach(entry -> {
+            for(TabulaAnimationComponentContainer component : entry.getValue()) {
+                applyComponent(component, model.getCubeByIdentifier(entry.getKey()), entity);
+            }
+        });
+
+        postAnimationCallback(model, entity, limbSwing, limbSwingAmount, ageInTicks, rotationYaw, rotationPitch);
+    }
+
+    /**
+     * Callback to use if you want to change model parts **before animating** but **after resetting to the default pose**. Uses the same argument as {@link #setRotationAngles(TabulaModel, Entity, float, float, float, float, float, float)}
+     */
+    protected void preAnimationCallback(TabulaModel model, T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float rotationYaw, float rotationPitch) { }
```

Comment:
The term callback seems odd here. May be wrong, but I'd imagine 'callback' would only be used in a situation where you were calling it directly.

Reply:
Hmm. Maybe `prepareRender()` and `finishRender()`?

Reply:
I chose `prepareAnimation` & `finishAnimation` because there is no render done in-between